crates/light_sdk/src/transaction/transfer.rs

```diff
@@ -1,46 +1,102 @@
-use borsh_ext::BorshSerializeExt;
 use namada_sdk::address::Address;
 use namada_sdk::hash::Hash;
 use namada_sdk::key::common;
-use namada_sdk::token::DenominatedAmount;
+use namada_sdk::token::transaction::Transaction;
+use namada_sdk::token::ShieldingTransferData;
+pub use namada_sdk::token::{
+    DenominatedAmount, TransparentTransfer, UnshieldingTransferData,
+};
 use namada_sdk::tx::data::GasLimit;
-use namada_sdk::tx::{Authorization, Tx, TxError};
+use namada_sdk::tx::{
+    Authorization, Tx, TxError, TX_SHIELDED_TRANSFER_WASM,
+    TX_SHIELDING_TRANSFER_WASM, TX_TRANSPARENT_TRANSFER_WASM,
+    TX_UNSHIELDING_TRANSFER_WASM,
+};
 
 use super::{attach_fee, attach_fee_signature, GlobalArgs};
 use crate::transaction;
 
-const TX_TRANSFER_WASM: &str = "tx_transfer.wasm";
-
 /// A transfer transaction
 #[derive(Debug, Clone)]
 pub struct Transfer(Tx);
 
 impl Transfer {
-    /// Build a raw Transfer transaction from the given parameters
-    pub fn new(
-        source: Address,
-        target: Address,
-        token: Address,
-        amount: DenominatedAmount,
-        // TODO(namada#2596): handle masp here
-        shielded: Option<Hash>,
+    /// Build a transparent transfer transaction from the given parameters
+    pub fn transparent(
+        transfers: TransparentTransfer,
         args: GlobalArgs,
     ) -> Self {
-        let init_proposal = namada_sdk::token::Transfer {
-            source,
-            target,
-            token,
-            amount,
-            shielded,
-        };
-
         Self(transaction::build_tx(
             args,
-            init_proposal.serialize_to_vec(),
-            TX_TRANSFER_WASM.to_string(),
+            transfers,
+            TX_TRANSPARENT_TRANSFER_WASM.to_string(),
         ))
     }
 
+    /// Build a shielded transfer transaction from the given parameters
+    pub fn shielded(
```

Comment:
how out of data is our c bindings example?

Reply:
I fear very much 😢